### ISSUE TITLE
Setters for readOnly and disableEditor are async operations

### ISSUE THREAD

**Issue by johnguirgis** [recogito/annotorious-openseadragon#56]
Outside of readOnly mode, I am able to create, edit, and delete annotations with no issues, but as soon as I set `readOnly` to true and attempt to create an annotation, this happens:
![annotation_issue](https://user-images.githubusercontent.com/62853290/120322085-05128880-c2b2-11eb-8f16-3c6acbb4bb05.gif)
Also, when I am logging `anno.readOnly` I am getting that it is undefined even though I am setting it on the previous line.
Here is my init code:
```javascript
var perms = drupalSettings.recogito_integration.permissions;
  var user_data = drupalSettings.recogito_integration.user_data;
  var readOnly1 = (!perms['recogito create annotations'] &&
      !perms['recogito edit annotations'] &&
      !perms['recogito delete annotations'] &&
      !perms['recogito edit own annotations'] &&
      !perms['recogito delete own annotations']);

  var image_anno = OpenSeadragon.Annotorious(viewer);
  image_anno.readOnly = (readOnly1 || window.location.search !== "?mode=annotation");
  image_anno.disableEditor = image_anno.readOnly;
  image_anno.setAuthInfo({'id': user_data.id, 'displayName': user_data.displayName});

  var page_url = window.location.pathname;

  jQuery.ajax({
    type: "GET",
    url: "/recogito_integration/get",
    dataType: 'json',
    headers: {
      'pageurl': page_url,
    },

    success: function (data) {
      console.log(data);
      for (annotation in data) {

        w3c = convert_annotation_w3c(data[annotation]);
        if(w3c.type != "Annotation")
        {
          image_anno.addAnnotation(w3c);
        }
      }
    },
    error: function (xhr, status, error) {
      alert(xhr.responseText);
    }
  });


  image_anno.on('selectAnnotation', function (annotation) {

    highlightAnnotatedContent(annotation);

  });

  image_anno.on('createAnnotation', function (annotation) {

    console.log("createAnnotation: " + typeof annotation);
    create_annotation(annotation);
  });

  image_anno.on('updateAnnotation', function (annotation, previous) {

    console.log(annotation);
    console.log(previous);
    update_annotation(annotation, previous);
  });

  image_anno.on('deleteAnnotation', function (annotation) {
    delete_annotation(annotation);
  });
```

**Comment by rsimon** [comment]
Thanks for the info! That's indeed weird. I can only guess that this might be some kind of race condition, where Anntorious is not yet ready at the time you're setting `readOnly`.

Since you are setting this essentially at init time, can you try the following alternative instead? 

```js
var image_anno = OpenSeadragon.Annotorious(viewer, {
  readOnly: readOnly1 || window.location.search !== "?mode=annotation",
  disableEditor: readOnly1 || window.location.search !== "?mode=annotation" 
});

image_anno.setAuthInfo({'id': user_data.id, 'displayName': user_data.displayName});
```

This should definitely make sure readOnly gets set initially.

**Comment by johnguirgis** [comment]
Works great now, thanks! :smiley:

**Comment by rsimon** [comment]
Great to hear this works. I'm keeping the issue open, because the getter/setter should be working immediately. (I do have a hunch what's happening & it may not be entirely fixable. However in this case I will at least document the issue.)

**Comment by rsimon** [comment]
BTW: it's really cool that you are using RecogitoJS and Annotorious in combination! Will this project become public and some way? And/or can you share the project goals somehow? I'm really interested in your use cases. And also if you see possibilities (or requirements) for linking between specific text and image annotations. (I know that some people have already experimented with that. But myself, I haven't really thought about it much yet.)

**Comment by johnguirgis** [comment]
Thanks for your interest. I am working on annotation modules that will eventually be used in [this](https://dragomans.digital.utsc.utoronto.ca/) project. As of right now, there is no requirement for linking between text and image annotations but I am sure that that will be the next step. I will definitely look into that though :smile:

**Comment by rsimon** [comment]
Looks great! (My day job is actually in the Digital Humanities, too :-) Looking forward to seeing this progress.

**Comment by rsimon** [comment]
FYI info there was an additional bug that cause the switching between read & write mode to break when the initial setting was `readOnly: true`. This is now fixed and will go into the next release (v2.4.1).
